embuilder.py

```diff
@@ -109,17 +102,21 @@
       '''
 
 SYSTEM_TASKS = [
-    'al', 'compiler-rt', 'gl', 'gl-mt', 'libc', 'libc-mt', 'libc-extras',
-    'emmalloc', 'emmalloc_debug', 'dlmalloc', 'dlmalloc_threadsafe', 'pthreads',
-    'dlmalloc_debug', 'dlmalloc_threadsafe_debug', 'libc++', 'libc++_noexcept',
-    'dlmalloc_debug_noerrno', 'dlmalloc_threadsafe_debug_noerrno',
-    'dlmalloc_noerrno', 'dlmalloc_threadsafe_noerrno',
+    'al', 'compiler-rt', 'gl', 'gl-mt', 'gl-emu', 'libc', 'libc-mt', 'libc-extras',
+    'emmalloc', 'emmalloc_debug', 'pthreads', 'libc++', 'libc++_noexcept',
     'libc++abi', 'html5'
 ]
+for debug in ['', '_debug']:
+  for noerrno in ['', '_noerrno']:
+    for threadsafe in ['', '_threadsafe']:
+      for tracing in ['', '_tracing']:
```

Comment:
For specializations that only affect one library such as dlmalloc, can we just have the user manually do `-ldlmalloc_tracing` instead of using a `-s` flag?

Reply:
Oh, actually nevermind that's orthogonal to this file which just controls which flavors get built.